spec/models/session_spec.rb

```diff
@@ -32,6 +32,17 @@
         expect(subject.ttl).to be_an(Integer)
         expect(subject.ttl).to be_between(0, 3600)
       end
+
+      it 'will extend a session within the maximum ttl' do
+        subject.created_at = subject.created_at - (described_class::MAX_SESSION_LIFETIME - 1.minute)
+        expect(subject.expire(3600)).to eq(true)
+      end
+
+      it 'will not extend a session beyond the maximum ttl' do
+        subject.created_at = subject.created_at - (described_class::MAX_SESSION_LIFETIME + 1.minute)
```

Comment:
When using Timecop to move ~12 hours into the future, fakeredis purges key from redis.  My original spec looked like this:
```ruby
        # keep extending session so Redis doesn't kill it
        increment = subject.redis_namespace_ttl - 1.minute
        max_hours = described_class::MAX_SESSION_LIFETIME / 1.hour
        (1..max_hours).each do |hour|
          Timecop.freeze(start_time + increment * hour)
          expect(subject.save).to eq(true)
          expect(subject.ttl).to eq(Session.redis_namespace_ttl)
        end
```

I think that just manually changing the subject.created_at accomplishes the same thing and is much simpler.